sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -164,4 +164,19 @@ case class HiveTableScanExec(
   }
 
   override def output: Seq[Attribute] = attributes
+
+  override def sameResult(plan: SparkPlan): Boolean = plan match {
+    case other: HiveTableScanExec =>
+      val thisPredicates = partitionPruningPred.map(cleanExpression)
+      val otherPredicates = other.partitionPruningPred.map(cleanExpression)
+
+      val result = relation.sameResult(other.relation) &&
+        output.length == other.output.length &&
+          output.zip(other.output)
+            .forall(p => p._1.name == p._2.name && p._1.dataType == p._2.dataType) &&
```

Comment:
For example, the full output of table is (a: Int, b: Int), and output1 is (a: Int), output2 is (b: Int), their dataType are same, but they are diff.